File: LR1/main.cpp
#include <iostream>
#include "./Utils/utils.hpp"
#include "./Utils/utils.cpp"
#include "./Data/data.hpp"
#include "./Gauss/gauss.hpp"


int main()
{
    const auto coefficients{kMatrixC * kOption + kMatrixD};
    const auto &free_coefficients{kVectorB};

    const auto response{SolveByGauss(coefficients, free_coefficients)};

    std::cout << "Solution: ";
    for (const auto &item : response)
        std::cout << item << ' ';

    return 0;
}

File: LR1/Py/main.py
import data
from utils import sum_two_matrices, multiply_matrix_by_ratio, get_copy
from gauss import gauss


def main() -> None:
    main_coefficients = sum_two_matrices(
        multiply_matrix_by_ratio(data.MATRIX_C, data.OPTION),
        data.MATRIX_D
    )

    free_coefficients = get_copy(data.VECTOR_B)

    solution = gauss(main_coefficients, free_coefficients)
    print(solution)


if __name__ == '__main__':
    main()
